x/incentive/msgs.go

```diff
@@ -0,0 +1,214 @@
+package incentive
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/umee-network/umee/v3/util/checkers"
+	leveragetypes "github.com/umee-network/umee/v3/x/leverage/types"
+)
+
+var (
+	_ sdk.Msg = &MsgClaim{}
+	_ sdk.Msg = &MsgBeginUnbonding{}
+	_ sdk.Msg = &MsgBond{}
+	_ sdk.Msg = &MsgSponsor{}
+	_ sdk.Msg = &MsgGovSetParams{}
+	_ sdk.Msg = &MsgGovCreateProgram{}
+)
+
+func NewMsgClaim(account sdk.AccAddress) *MsgClaim {
+	return &MsgClaim{
+		Account: account.String(),
+	}
+}
+
+func (msg *MsgClaim) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Account)
+	return err
+}
+
+func (msg *MsgClaim) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Account)
+}
+
```

Comment:
Can you implement `Route` and `Type` methods for all messages , For Ledger support we need to implement this two methods
```go
// Route implements the sdk.Msg interface.
func (m MsgType) Route() string { return types.RouterKey }

// Type implements the sdk.Msg interface.
func (m MsgType) Type() string { return sdk.MsgTypeURL(&m) }
```

Reply:
Got it